torchdata/nodes/samplers/utils.py

```diff
@@ -0,0 +1,24 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+
+class StopCriteria:
```

Comment:
Should we have a CYCLE_FOREVER option that just continuously restarts iterators until external stop signal? ie for training for fixed number of steps

Reply:
Can add later in a separate PR